pkg/reconciler/status.go

```diff
@@ -116,9 +122,21 @@ func (s *Status) SetReconcileCompleted(err error) {
 	s.UpdateFunc(s.S)
 }
 
-func (s *Status) SetDeleteCompleted(err error) {
-	s.removeAllConditions()
+// SetPackageYanked appends a new condition to indicate a package is yanked
+func (s *Status) SetPackageYanked(meta metav1.ObjectMeta, reason string) {
+	s.markObservedLatest(meta)
+
+	s.S.Conditions = append(s.S.Conditions, kcv1alpha1.Condition{
+		Type:   kcv1alpha1.PackageYanked,
+		Status: corev1.ConditionTrue,
+		Reason: reason,
+	})
 
+	s.UpdateFunc(s.S)
```

Comment:
i would also try to avoid making an additional api request for non-terminal conditions

Reply:
I think we have to make the api request to update the status of the PackageInstall right? I'm a little confused by how we can get it to update without the api call

Reply:
@cppforlife